PodongPodong/PodongPodong/Presentations/Search/ViewModels/SearchViewModel.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  SearchViewModel.swift
+//  PodongPodong
+//
+//  Created by Paidion on 6/4/25.
+//
+
+import Foundation
+import Combine
+
+final class SearchViewModel: ObservableObject {
+    var selectedTab: PartyListTab = .공동구매
+
+    @Published var searchText: String = ""
+    @Published var isLoading: Bool = false
+    @Published var isShowingFilterView: Bool = false
+
+    // MARK: 필터
+    @Published var isShowingNotCompletedPartyOnly: Bool = false
+    @Published var foodCategory: FoodCategory? = nil
+    @Published var purchaseChannel: PurchaseChannel? = nil
+
+    var isFilterApplied: Bool {
+        foodCategory != nil
+        || purchaseChannel != nil
+    }
+
+    @Published private(set) var partyList: [Party] = []
+
+    private var cancellables = Set<AnyCancellable>()
+
+    init() {
+        Publishers.CombineLatest4($isShowingNotCompletedPartyOnly, $foodCategory, $purchaseChannel, $searchText)
+            .debounce(for: .milliseconds(300), scheduler: RunLoop.main)
+            .sink { [weak self] _, _, _, _ in
+                Task { await self?.fetchPartyList() }
+            }
+            .store(in: &cancellables)
+    }
+
+    @MainActor
+    func fetchPartyList() async {
+        do {
+            let parties: [Party] = try await FirestoreManager.shared.fetch(
+                as: Party.self,
+                .party,
+                whereFeild: "title",
+                equalData: searchText.isEmpty ? nil : searchText,
+//                count: 0
+            )
+            self.partyList = parties
+            print("====================")
+            print(parties)
+            print("====================")
+        } catch {
+            print("파티 목록을 불러오는데 실패했습니다: \(error.localizedDescription)")
+        }
+    }
```

Comment:
```suggestion
    func fetchPartyList() async {
        Task {
            do {
                let parties: [Party] = try await FirestoreManager.shared.fetch(
                    as: Party.self,
                    .party,
                    whereFeild: "title",
                    //                equalData: searchText.isEmpty ? nil : searchText,
                    //                count: 0
                )
                self.partyList = parties
                print("====================")
                print(parties)
                print("====================")
            } catch {
                print("파티 목록을 불러오는데 실패했습니다: \(error.localizedDescription)")
            }
        }
    }
```

이렇게 쓰면 더 좋을 것 같아요

메인 엑터 - 메인 큐로 UI업데이트 담당
FireStoreManager 를 Task로 감싸서 비동기 작업을 다른 큐에 시킴

이렇게 이해하면 될 것 같아요 궁금한건 나중에 만나서 더 질문하셔도 됩니당

Reply:
오! 무슨 말씀인지 아직 정확하게 이해는 못했는데, 공부해보고 작업해보도록 하겠습니다!

Reply:
현재 커밋(저가 보낸)같이 하면 상태변화의 경우 메인액터로 Task밖에 빼두고
Task를 통해서(다른 큐) Firestroe관련된 부분이 변경되야 partyList가 수정되는 부분이 맞아요!

그래서 await MainActor.run은 만약에 Firestore에서 데이터 에러가 걸리거나 송수신이 느려지면 mainActor가 멈추는건 치명적이기 때문에 위와 같이 하는게 좋다고 했습니다!